lib/sycamore/sycamore/scans/db_scan.py

```diff
@@ -0,0 +1,40 @@
+from opensearchpy import OpenSearch
+from ray.data import Dataset, from_items
+
+from sycamore.data import Document
+from sycamore.plan_nodes import Scan
+
+
+class OpenSearchScan(Scan):
+    def __init__(
+        self,
+        index_name: str,
+        os_client_args: dict,
+        query=None,
+        **resource_args,
+    ):
+        super().__init__(**resource_args)
+        self.index_name = index_name
+        self.os_client_args = os_client_args
+        self.os_client = OpenSearch(**os_client_args)
+        self.query = query
+
+    def execute(self) -> Dataset:
+        result = []
+
+        scroll = "1m"
+        response = self.os_client.search(index=self.index_name, scroll=scroll, size=200, body=self.query)
+        scroll_id = response["_scroll_id"]
+
+        while True:
+            hits = response["hits"]["hits"]
+            for hit in hits:
+                result += [Document(hit["_source"])]
+
+            if not hits:
+                break
+            response = self.os_client.scroll(scroll_id=scroll_id, scroll=scroll)
```

Comment:
Good question, I don't know either

Reply:
Please then do a manual test with the timeout set to 1s so that the total download takes several seconds. If it returns the same thing then it's extending the timeout.  My guess is that it is, but I couldn't find anything definitive.

Reply:
Validated, added a 0.1s sleep between each call, reduced the size for each call to 1 record, and used a timeout of 1s. Works fine and took ~1m to fully run.